copycat/file/src/main/java/org/apache/kafka/copycat/file/FileStreamSourceTask.java

```diff
@@ -52,22 +58,29 @@ public void start(Properties props) {
         } else {
             try {
                 stream = new FileInputStream(filename);
-                Long lastRecordedOffset = (Long) context.getOffsetStorageReader().getOffset(null);
-                if (lastRecordedOffset != null) {
-                    log.debug("Found previous offset, trying to skip to file offset {}", lastRecordedOffset);
-                    long skipLeft = lastRecordedOffset;
-                    while (skipLeft > 0) {
-                        try {
-                            long skipped = stream.skip(skipLeft);
-                            skipLeft -= skipped;
-                        } catch (IOException e) {
-                            log.error("Error while trying to seek to previous offset in file: ", e);
-                            throw new CopycatException(e);
+                SchemaAndValue offsetWithSchema = context.offsetStorageReader().offset(new SchemaAndValue(OFFSET_KEY_SCHEMA, filename));
```

Comment:
One thing I'm not happy with here is the `SchemaAndValue` class. Besides the kind of awkward name, I'd kind of like to avoid exposing the user to it, but it's necessary for return values. One thing I'm wondering is if it would be preferable to always make method parameters take the individual parts (separate args for schema and value) or if the symmetry of using SchemaAndValue in an interface for both args and return value is better.